client/my-sites/post-type-list/post-actions-ellipsis-menu/copy-link.jsx

```diff
@@ -0,0 +1,62 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize, translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import PopoverMenuItemClipboard from 'components/popover/menu-item-clipboard';
+import { getPost } from 'state/posts/selectors';
+import { bumpStat, recordTracksEvent } from 'state/analytics/actions';
+import { bumpStatGenerator } from './utils';
+import { infoNotice } from 'state/notices/actions';
+
+function PostActionsEllipsisMenuCopyLink( { onCopyLinkClick, copyLink } ) {
+	return (
+		<PopoverMenuItemClipboard text={ copyLink } onCopy={ onCopyLinkClick } icon={ 'link' }>
+			{ translate( 'Copy Link' ) }
+		</PopoverMenuItemClipboard>
+	);
+}
+
+PostActionsEllipsisMenuCopyLink.propTypes = {
+	onCopyLinkClick: PropTypes.func,
+	copyLink: PropTypes.string,
+	translate: PropTypes.func,
+};
+
+const mapStateToProps = ( state, { globalId } ) => {
+	const post = getPost( state, globalId );
+	if ( ! post ) {
+		return {};
+	}
+
+	return {
+		copyLink: post.URL,
+		type: post.type,
```

Comment:
Optional, but we do have a lodash `get` utility for this like `copyLink: get( post, 'URL', null )` or a simple check if we're afraid of performance impact: `copyLink: post ? post.URL : null`

Reply:
Hopefully we'll get to use optional chaining soon. https://github.com/tc39/proposal-optional-chaining

Reply:
Do those help anything? We already know `post` is truthy, so the worst we'll get is a value of `undefined` (no reference errors). Maybe `null` would be better than `undefined` (although I don't see a difference with what they're used for).

Reply:
Mainly I'd prefer the inline check if we added more props here. This is why it's optional.